### DISCUSSION TITLE
How to use the "github_auth" feature?

### DISCUSSION THREAD

**Discussion by chunjiw** [coder/code-server#5513] [Q&A] [answered]
I noticed in the [Change log](https://github.com/coder/code-server/blob/33ee184ed7b168acf8b68d89e6ded3dd80ad80d8/CHANGELOG.md#410---2022-03-03) that I can set GITHUB_TOKEN environment variable. Is it true that after setting this env we can directly `git clone` a private repo? Because when I tried it, vscode still asks me to login to GitHub, either through device auth or username/password. 

What is the correct way to use github_auth? Thank you!

**Comment by jsjoeio** [comment, marked as answer]
Interesting  🤔 I just helped someone else with this. After setting your `GITHUB_TOKEN`, you source'd your shell, ensured it was set and then tried again?

Did you try restarting code-server?

It should _just work_. This is what our e2e test does to check if it's working: https://github.com/coder/code-server/blob/main/test/e2e/github.test.ts#L5-L17

**Reply by chunjiw** [reply]
Thank you for your reply! Didn't get a chance to dig more into it this week, will get back to it asap this weekend :)

**Reply by chunjiw** [reply]
I tried newest version 4.6.1 and it still didn't work. I set PASSWORD successfully.. but git still asks for auth. Using git ssh leads to "permission denied".

I confirmed that the env is set correctly since I can echo both PASSWORD and GITHUB_TOKEN... I'm quite confused, I must have done something wrong that I don't know of.

**Reply by jsjoeio** [reply]
interesting! thanks for trying that. @code-asher any ideas?

**Reply by code-asher** [reply]
How are you doing the `git clone`?  Are you using the `Git: Clone` command, through the UI, or on the command line?  code-server reads the environment variable but I do not think `git` does so running `git` on the command line is unaffected.  In that case you would need to set up `git` to use the token.

**Reply by chunjiw** [reply]
I see! I'm not using the UI, just `git clone` in the terminal. You explained very well, thank you! Gonna try the UI `Git: Clone` command.

**Comment by chunjiw** [comment]
Some more details: I'm using Kubernetes to deploy code-server in a pod like so:
```
      containers:
        - name: backend
          image: ghcr.io/myname/myrepo:main
          imagePullPolicy: Always
          command: ["/bin/bash"]
          args:
            - -c
            - |
              echo "password is $PASSWORD and git token is $GITHUB_TOKEN";
              code-server --install-extension golang.go;
              code-server --disable-telemetry --disable-update-check --bind-addr 0.0.0.0:8002
          env:
            - name: PASSWORD
              value: {{.Values.backend_access_password}}
            - name: GITHUB_TOKEN
              value: {{.Values.github_token}}
```
